### ISSUE TITLE
Add a parameter to provide a .sqlfluff configuration file when using the .fix or .lint API methods

### ISSUE THREAD

**Issue by mayconrpgm** [sqlfluff/sqlfluff#1274]
Currently, when we use the sqlfluff API on a Python script as described in https://docs.sqlfluff.com/en/stable/api.html, it doesn't allow you to specify a configuration file built using the instructions at https://docs.sqlfluff.com/en/stable/configuration.html, it will only do the linting and fixing of SQL statements using the default configurations, allowing only to specify which dialect will be used:

```
import sqlfluff

sql_string = "SeLEct  *, 1, blah as  fOO  from myTable"

lint_result = sqlfluff.lint(sql_string, dialect="postgres")
print(lint_result)

fix_result = sqlfluff.fix(sql_string, dialect="postgres")
print(fix_result)
```

I was able to implement this funcionality by using the internal API as follows:

```
sql_string = "SeLEct  *, 1, blah as  fOO  from myTable"

parsed_sql = sqlfluff.parse(sql_string)

fluff_config = FluffConfig.from_path(path='.sqlfluff')
fluff_linter = Linter(config=fluff_config)

fixed_sql, linter_result = fluff_linter.fix(tree=parsed_sql.tree, config=fluff_config)

print(fixed_sql.raw)
```

But it's a little more cumbersome to use and I had to take some time digging into the code to better understand how to get the information needed from the returned objects.

It would be nice if the `sqlfluff.lint` and `sqlfluff.fix` functions had an optional parameter to pass a path to a configuration file or even a FluffConfig object by using the `FluffConfig.from_path` method from above:

```
import sqlfluff

sql_string = "SeLEct  *, 1, blah as  fOO  from myTable"

lint_result = sqlfluff.lint(sql_string, config="<path_to_sqlfluff_config_file>")
print(lint_result)

fix_result = sqlfluff.fix(sql_string, config="<path_to_sqlfluff_config_file>")
print(fix_result)
```

This issue is similar to this one, only that it applies to the API instead of the CLI: #1244

**Comment by andres-lowrie** [comment]
don't want to hijack this thread but I think this is the correct place for these comments given the linked issues.

It would be helpful to be able to pass in an argument to the cli to have it override the various directories it looks for config files for something like 

```
sqlfluff --path-to-config-file some/path/here
```

The use-case that I have is that I want to store the config file in a repository and the then run linting in a CI/CD pipeline ... without having to cp files into `$HOME` dirs or do much outside of `pip install`ing the tool.

Basically have a way to tell this function 
https://github.com/sqlfluff/sqlfluff/blob/117740cfcc3101e2dfcb6674776611dfad343707/src/sqlfluff/core/config.py#L351

to use the command line argument first and then go through the hierarchy.

**Comment by jpy-git** [comment]
@andres-lowrie what you describe has already been done and is in the main branch and will be included in the next release this weekend (See #1986 and #2061). I'm also just adding the Simple API option now 👍
